tests/ui/format_args.rs

```diff
@@ -97,9 +98,17 @@ fn main() {
     println!("{}", Z(1).to_string());
     println!("{}", x.to_string());
     println!("{}", x_ref.to_string());
+    println!("{} and again {0}", x.to_string());
+    // https://github.com/rust-lang/rust-clippy/issues/7903
+    println!("{foo}{foo}", foo = "foo".to_string());
+    println!("{foo}{bar}", foo = "foo".to_string(), bar = "bar");
+    println!("{foo}{bar}", foo = "foo", bar = "bar".to_string());
+    println!("{foo}{bar}", bar = "bar".to_string(), foo = "foo");
+    println!("{foo}{bar}", bar = "bar", foo = "foo".to_string());
 
     println!("error: something failed at {}", Somewhere.to_string());
-    println!("{} and again {0}", x.to_string());
```

Comment:
See my other comment on this. Comments would be helpful here. One to separate `// negative tests` and one to explain the intent behind this test since it isn't obvious.